### PR TITLE
Implement readFile

### PR DESCRIPTION
**Pull request by kevinkassimo**
Implement async version for readFile. Simulated `fetch` in its handling
May have problem with overflowing ids (to be fixed along with that of `fetch` in the future)

### CODE REVIEW

File: os.ts [hunk 1]
```diff
@@ -63,3 +75,64 @@ export function writeFileSync(
     writeFileSyncPerm: perm
   });
 }
+
+const readFileRequests = new Map<number, ReadFileRequest>();
+
+// TODO replace with bigint
+let nextReadFileRequestId = 0;
```
Comment (YurySolovyov) [inline os.ts]: or use `UUID` of some sort?

Comment (kevinkassimo) [reply os.ts]: Still have possibility of collision, though seems better for reusing if old one is released. Will see if I could introduce a UUID implementation.

Comment (kevinkassimo) [reply os.ts]: Actually, I could just probably simple generate a random number below certain size (so I could easily pass to Go) and simply check if the id is already occupied and regenerate

**Comment by ry** [comment]
@kevinkassimo Thank you for this work - but things are very unstable with the rewrite to Rust - and I can't use this patch now. Let's revisit in a few days. Sorry about that. Closing this PR.

**Pull request closed without merging**